data/Link2.php

```diff
@@ -120,7 +120,7 @@ public function __construct($linkName, $bean, $linkDef = array())
         // Fix to restore functionality from Link.php that needs to be rewritten but for now this will do.
         $this->relationship_fields = (!empty($this->def['rel_fields'])) ? $this->def['rel_fields'] : array();
 
-        if (!$this->loadedSuccesfully()) {
+        if (!$this->loadedSuccesfully() && (!isset($this->def['source']) || $this->def['source'] !== 'non-db')) {
```

Comment:
This is too broad of a spectrum.  There are hundreds of relationships that are non-db.  There is something wrong with this relationship hence why it isn't loading in RelationshipFactory